portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -28,51 +31,75 @@
 import java.util.ArrayList;
 
 
-/** Servlet that returns some example content. TODO: modify this file to handle comments data */
+/** Servlet that processes comments and analyses comment sentiment. */
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
   Gson gson;
   DatastoreService datastore;
+  LanguageServiceClient languageService;
 
-  public DataServlet() {
+  public DataServlet() throws IOException {
     gson = new Gson();
     datastore = DatastoreServiceFactory.getDatastoreService();
+    languageService = LanguageServiceClient.create();
   }
 
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {  
     // Get comments from Datastore and show on the page.
-    ArrayList<String> messages = new ArrayList<String>();
-    Query query = new Query("Comment").addSort("text", SortDirection.DESCENDING);
+    ArrayList<Comment> comments = new ArrayList<Comment>();
+    Query query = new Query("Comment").addSort("sentiment", SortDirection.DESCENDING);
     PreparedQuery results = datastore.prepare(query);
 
     for (Entity entity : results.asIterable()) {
-        String comment = (String) entity.getProperty("text");
-        messages.add(comment);
+        String comment_text = (String) entity.getProperty("text");
+        double sentimentScore = (double) entity.getProperty("sentiment");
+        Comment comment = new Comment(comment_text, sentimentScore);
+        comments.add(comment);
     }
 
-    String json = convertToJsonUsingGson(messages);
+    String json = convertToJsonUsingGson(comments);
     response.setContentType("application/json;");
     response.getWriter().println(json);
   }
 
   /**
-   * Converts a Java ArrayList<String> into a JSON string using the Gson library.
+   * Converts a Java ArrayList<Comment> into a JSON string using the Gson library.
    */
-  private String convertToJsonUsingGson(ArrayList<String> listOfStrings) {
+  private String convertToJsonUsingGson(ArrayList<Comment> listOfStrings) {
```

Comment:
Rename commentList? Also didn’t know that gson could convert structured data too, that’s cool!